tests_integration/test_main.py

```diff
@@ -542,3 +545,73 @@ def test_protocols_udp_restricted_port_values_start_must_be_lte_end(
 
         stderr = ex.value.stderr.decode()
         assert "Start port value must be less or equal to end port value" in stderr
+
+
+@pytest.mark.integration()
+def test_resource_access_flows(kopf_settings, unique_resource_name):
+    assert "TWINGATE_PRINCIPAL_ID" in os.environ
+    principal_id = os.environ["TWINGATE_PRINCIPAL_ID"]
+
+    RESOURCE_OBJ = f"""
+        apiVersion: twingate.com/v1beta
+        kind: TwingateResource
+        metadata:
+          name: {unique_resource_name}
+        spec:
+          name: My K8S Resource
+          address: my.default.cluster.local
+    """
+
+
+    OBJ = f"""
+        apiVersion: twingate.com/v1beta
+        kind: TwingateResourceAccess
+        metadata:
+          name: {unique_resource_name}
+        spec:
+          resourceRef:
+            name: {unique_resource_name}
+          principalId: {principal_id}
+    """
+
+    # fmt: off
+    with KopfRunner(kopf_runner_args, settings=kopf_settings) as runner:
+        kubectl_create(RESOURCE_OBJ)
+        time.sleep(10)  # give it some time to react
```

Comment:
I dont think we need 2 sleeps, we can call `kubectl_create` twice then sleep for `5s`

Reply:
I tried with 5s sleep but the resourceAccess creating was failing (error was the resource not created yet)